### PR TITLE
Secondary storage: Allow store deletion after successful data migration

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description

This PR fixes the issue reported in :https://github.com/apache/cloudstack/issues/4613
This issue seems to have cropped up due to code revert that happened by commit - d864e9dc39b69e590e3df898a03a874a8173fcc6

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?
1. Migrate data from one image store to another
2. Attempt deleting the store from where all data objects we migrated

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan package

**Review by harikrishna-patnala — APPROVED** [review]
LGTM

File: server/src/main/java/com/cloud/api/query/dao/TemplateJoinDaoImpl.java [hunk 1]
```diff
@@ -522,6 +522,7 @@ public TemplateResponse newIsoResponse(TemplateJoinVO iso) {
         sc.setParameters("store_id", storeId);
         sc.setParameters("type", TemplateType.USER);
         sc.setParameters("templateState", VirtualMachineTemplate.State.Active);
+        sc.setParameters("public", Boolean.FALSE);
```
Comment (shwstppr) [inline server/src/main/java/com/cloud/api/query/dao/TemplateJoinDaoImpl.java:525]: @Pearl1594 why this change is needed?

Comment (Pearl1594) [reply server/src/main/java/com/cloud/api/query/dao/TemplateJoinDaoImpl.java:525]: @shwstppr Deletion of image stores will be prevented only if there are private user templates present on the respective store.

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Review by rhtyd — APPROVED** [review]
